samples/hello-cube.html

```diff
@@ -0,0 +1,336 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta name="viewport" content="width=600">
+<title>WebGPU Cube demo</title>
+<script src="gl-matrix-min.js"></script>
+</head>
+<body>
+<canvas height=1000 width=1000></canvas>
+<script>
+const shader = `
+#include <metal_stdlib>
+
+using namespace metal;
+
+struct Vertex
+{
+    float4 position [[attribute(0)]];
+    float4 color [[attribute(1)]];
+};
+
+struct FragmentData
+{
+    float4 position [[position]];
+    float4 color;
+};
+
+struct Uniform
+{
+    device float4x4* modelViewProjectionMatrix [[id(0)]];
+};
+
+vertex FragmentData vertex_main(Vertex vertexIn [[stage_in]],
+                          const device Uniform& uniforms [[buffer(1)]])
+{
+    FragmentData output;
+    output.position = uniforms.modelViewProjectionMatrix[0] * vertexIn.position;
+    output.color = vertexIn.color;
+
+    return output;
+}
+
+fragment float4 fragment_main(FragmentData data [[stage_in]])
+{
+    return float4(data.color);
+}
+`
+
+// A * indicates that this symbol has yet to be implemented in WebKit.
+
+let device, context, verticesBuffer, mappedTransformArray, uniformsBindGroup, pipeline, renderPassDescriptor, queue;
+let projectionMatrix = mat4.create();
+
+const vertexSize = 4 * 8;
+const verticesSize = vertexSize * 36;
+const colorOffset = 4 * 4;
+function populateVerticesArray(array) {
+    const vertexData = [
+        // float4 position, float4 color
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+    ];
```

Comment:
This could be stored as a Float32Array, and populateVertexArray would just become `mappedArray.set(vertexData, 0);`